source/Api.cpp

```diff
@@ -46,21 +54,53 @@ namespace Aws
         static void s_cleanUpApi()
         {
             g_allocator = nullptr;
+            aws_auth_library_clean_up();
             aws_mqtt_library_clean_up();
             aws_http_library_clean_up();
             Io::CleanUpTlsStaticState();
         }
 
-        ApiHandle::ApiHandle(Allocator *allocator) noexcept { s_initApi(allocator); }
+        ApiHandle::ApiHandle(Allocator *allocator) noexcept : logger() { s_initApi(allocator); }
+
+        ApiHandle::ApiHandle() noexcept : logger() { s_initApi(DefaultAllocator()); }
 
-        ApiHandle::ApiHandle() noexcept { s_initApi(DefaultAllocator()); }
+        ApiHandle::~ApiHandle()
+        {
+            s_cleanUpApi();
 
-        ApiHandle::~ApiHandle() { s_cleanUpApi(); }
+            if (aws_logger_get() == &logger)
+            {
+                aws_logger_set(NULL);
+                aws_logger_clean_up(&logger);
+            }
+        }
 
-        void LoadErrorStrings() noexcept
+        void ApiHandle::InitializeLogging(Aws::Crt::LogLevel level, const char *filename)
```

Comment:
it might be good to accept a file pointer as well so we can dump to stderr or stdout etc....

Reply:
Added an overload that takes a file pointer and refactored to make nicer.